types.go

```diff
@@ -108,20 +108,16 @@ type FunctionInfo struct {
 
 // functionInfo contains information about which interfaces the objective
 // function F implements and the actual methods of F that have been
-// successfully type switched.
+// successfully type switched. TODO(jds) change this comment
 type functionInfo struct {
```

Comment:
In any case, none of the current optimizers actually need FunctionInfo that we pass to it.